contrib/nydusify/pkg/chunkdict/generater/generater.go

```diff
@@ -0,0 +1,117 @@
+package generater
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/dragonflyoss/image-service/contrib/nydusify/pkg/parser"
+	"github.com/dragonflyoss/image-service/contrib/nydusify/pkg/provider"
+	"github.com/dragonflyoss/image-service/contrib/nydusify/pkg/utils"
+)
+
+// Opt defines Chunkdict generate options.
+// Note: sources is one or more Nydus image references.
+type Opt struct {
+	WorkDir        string
+	Sources        string
+	SourceInsecure bool
+	NydusImagePath string
+	ExpectedArch   string
+}
+
+// "Chunkdict generate" command store multiple Nydus images into database
```

Comment:
`chunkdict generate` should be used for generating the chunk dict image directly.